src/components/data-loader.tsx

```diff
@@ -82,6 +80,8 @@ const withDataLoader = (BaseComponent: FC<{ data: unknown[] }>) => {
       handleAskForMoreData();
     };
 
+    const { current: ioSupport } = useRef('IntersectionObserver' in window);
```

Comment:
If feel this got moved here because of limitations in mocking in the tests.
I'd say in this case just replace both uses of `ioSupport` with the actual `'IntersectionObserver' in window` statement.

Reply:
Updated.